x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -402,11 +404,15 @@ public void analyzedPlan(
                     preAnalysis.inferencePlans,
                     l.map(preAnalysisResult::withInferenceResolution)
                 )
-            );
-        // first resolve the lookup indices, then the main indices
-        for (var index : preAnalysis.lookupIndices) {
-            listener = listener.andThen((l, preAnalysisResult) -> preAnalyzeLookupIndex(index, preAnalysisResult, executionInfo, l));
-        }
+            )
+            // first resolve the lookup indices, then the main indices
+            .<PreAnalysisResult>andThen((l, preAnalysisResult) -> {
```

Comment:
A comment wouldn't hurt to make more clear that this runs for all lookup indices in parallel.